src/main/kotlin/com/controller/DriverController.kt

```diff
@@ -0,0 +1,36 @@
+package com.controller
+
+import com.dto.ReservationManager
+import com.dto.ReservationRequest
+import com.dto.StartedReservationDto
+import com.dto.StoppedReservationDto
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.math.BigDecimal
+import java.time.LocalDateTime
+import javax.validation.Valid
+
+@RestController
+@RequestMapping("/driver/reservation")
+class DriverController(private val reservationManager: ReservationManager) {
+
+    @PostMapping("/start")
+    private fun startReservation(@Valid @RequestBody reservationRequest: ReservationRequest): StartedReservationDto =
+            reservationManager.createReservation(reservationRequest, LocalDateTime.now())
+
+
+    @PutMapping("/stop")
+    private fun stopReservation(@RequestParam reservationId: Long): StoppedReservationDto =
```

Comment:
czy reservatopm jest zasobem RESTowym? Jeśli tak, to jak lepiej ten numer mógł by być przesłany po sieci?

Reply:
hmm na pewno reservation jest zasobem więc może faktycznie konstrukcja powinna być w stylu reservation/{id}/stop